src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/clusterclient/WildcardFragment.kt

```diff
@@ -76,44 +106,74 @@ class WildcardFragment : Fragment() {
           val attributeName = ChipIdLookup.attributeIdToName(clusterId, attributeId)
           stringBuilder.append("\t\t$attributeName: ${attributeState.value}\n")
         }
+        clusterState.eventStates.forEach { (eventId, eventState) ->
+          val eventName = ChipIdLookup.eventIdToName(clusterId, eventId)
+          stringBuilder.append("\t\t$eventName: ${eventState.value}\n")
+        }
         stringBuilder.append("\t}\n")
       }
       stringBuilder.append("}\n")
     }
     return stringBuilder.toString()
   }
 
-  private suspend fun subscribe(minInterval: Int, maxInterval: Int) {
+  private suspend fun subscribe(type: Int, minInterval: Int, maxInterval: Int) {
     val subscriptionEstablishedCallback =
       SubscriptionEstablishedCallback { Log.i(TAG, "Subscription to device established") }
 
+    val resubscriptionAttemptCallback =
+      ResubscriptionAttemptCallback { terminationCause, nextResubscribeIntervalMsec
+                                     -> Log.i(TAG, "ResubscriptionAttempt terminationCause:$terminationCause, nextResubscribeIntervalMsec:$nextResubscribeIntervalMsec") }
+
     val endpointId = getChipPathIdForText(endpointIdEd.text.toString())
     val clusterId = getChipPathIdForText(clusterIdEd.text.toString())
     val attributeId = getChipPathIdForText(attributeIdEd.text.toString())
-    val attributePath = ChipAttributePath.newInstance(endpointId, clusterId, attributeId)
-
-    deviceController.subscribeToPath(subscriptionEstablishedCallback,
-                                     reportCallback,
-                                     ChipClient.getConnectedDevicePointer(requireContext(),
-                                                                          addressUpdateFragment.deviceId),
-                                     listOf(attributePath),
-                                     minInterval,
-                                     maxInterval)
+    val eventId = getChipPathIdForText(eventIdEd.text.toString())
+
+    if (type == ATTRIBUTE) {
+      val attributePath = ChipAttributePath.newInstance(endpointId, clusterId, attributeId)
+      deviceController.subscribeToPath(subscriptionEstablishedCallback,
+                                       reportCallback,
+                                       ChipClient.getConnectedDevicePointer(requireContext(),
+                                       addressUpdateFragment.deviceId),
+                                       listOf(attributePath),
+                                       minInterval,
+                                       maxInterval)
+    } else if (type == EVENT) {
```

Comment:
we really should consider to support event and attribute at the same time, in real situation, the chance is to subscribe/read wildcard attribute and event path.

Reply:
https://github.com/project-chip/connectedhomeip/issues/19671

Reply:
@joonhaengHeo